extensions/json-jackson/src/main/java/io/deephaven/json/jackson/TypedObjectMixin.java

```diff
@@ -184,73 +232,76 @@ public void processCurrentValue(JsonParser parser) throws IOException {
-            if (!allowMissing()) {
-                throw Parsing.mismatchMissing(parser, Object.class);
+            checkMissingAllowed(parser);
+            typeChunk.add(options.onMissing().orElse(null));
+            for (WritableChunk<?> sharedChunk : sharedChunks) {
+                addNullValue(sharedChunk);
             }
-            // onMissingType()?
-            typeOut.add(null);
-            for (Processor processor : processors.values()) {
+            for (Processor processor : combinedProcessors.values()) {
                 processor.notApplicable();
             }
         }
 
         private void processNullObject(JsonParser parser) throws IOException {
-            if (!allowNull()) {
-                throw Parsing.mismatch(parser, Object.class);
+            checkNullAllowed(parser);
+            typeChunk.add(options.onNull().orElse(null));
+            for (WritableChunk<?> sharedChunk : sharedChunks) {
+                addNullValue(sharedChunk);
             }
-            // onNullType()?
-            typeOut.add(null);
-            for (Processor processor : processors.values()) {
+            for (Processor processor : combinedProcessors.values()) {
                 processor.notApplicable();
             }
         }
 
         private void processEmptyObject(JsonParser parser) throws IOException {
-            // this logic should be equivalent to processObjectFields w/ no fields
-            // suggests that maybe this branch should be an error b/c we _need_ type field?
-            throw new IOException("no field");
+            throw new ValueAwareException("Expected a non-empty object", parser.currentLocation(), options);
         }
 
         private void processObjectFields(JsonParser parser) throws IOException {
             final String typeFieldValue = parseTypeField(parser);
-            if (!options.allowUnknownTypes()) {
-                if (!processors.containsKey(typeFieldValue)) {
-                    throw new IOException(String.format("Unmapped type '%s'", typeFieldValue));
-                }
-            }
-            typeOut.add(typeFieldValue);
-            if (parser.nextToken() == JsonToken.END_OBJECT) {
-                for (Processor processor : processors.values()) {
-                    processor.notApplicable();
-                }
-                return;
-            }
-            if (!parser.hasToken(JsonToken.FIELD_NAME)) {
-                throw new IllegalStateException();
-            }
-            for (Entry<String, Processor> e : processors.entrySet()) {
+            typeChunk.add(typeFieldValue);
+            parser.nextToken();
+            boolean foundProcessor = false;
+            for (Entry<String, Processor> e : combinedProcessors.entrySet()) {
                 final String processorType = e.getKey();
                 final Processor processor = e.getValue();
                 if (processorType.equals(typeFieldValue)) {
```

Comment:
Worth pointing out that we are not supporting aliases or case insensitivity here.